internal/charmstore/debug.go

```diff
@@ -14,60 +14,304 @@ import (
 	"strings"
 	"time"
 
+	"github.com/juju/httprequest"
 	"github.com/juju/utils"
+	"github.com/juju/utils/debugstatus"
+	"github.com/julienschmidt/httprouter"
 	"gopkg.in/errgo.v1"
 	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
 	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
 
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
 	"gopkg.in/juju/charmstore.v5-unstable/internal/router"
 	appver "gopkg.in/juju/charmstore.v5-unstable/version"
 )
 
-// GET /debug/info .
-func serveDebugInfo(http.Header, *http.Request) (interface{}, error) {
-	return appver.VersionInfo, nil
+type debugParams struct {
+	ServerParams
+	pool            *Pool
+	loopbackHandler http.Handler
 }
 
-// GET /debug/check.
-func debugCheck(checks map[string]func() error) http.Handler {
-	return router.HandleJSON(func(http.Header, *http.Request) (interface{}, error) {
-		n := len(checks)
-		type result struct {
-			name string
-			err  error
+type debugHandler struct {
+	router    *httprouter.Router
+	params    debugParams
+	checks    map[string]func() error
```

Comment:
Let's have some comments saying what these fields do. It's not
clear how or why the checks here differ from the ones used
for servicing debug/status, or what openPaths is for.